thrift/shared.thrift

```diff
@@ -1713,4 +1713,100 @@ struct RefreshWorkflowTasksRequest {
 
 struct FeatureFlags {
 	10: optional bool WorkflowExecutionAlreadyCompletedErrorEnabled
-}
\ No newline at end of file
+}
+
+enum CrossClusterTaskType {
+  StartChildExecution
+  CancelExecution
+  SignalExecution
+}
+
+enum CrossClusterTaskFailedCause {
+  DOMAIN_NOT_ACTIVE
+  WORKFLOW_ALREADY_RUNNING
+  WORKFLOW_NOT_EXISTS
+}
+
+struct CrossClusterTaskInfo {
+  10: optional string domainID
+  20: optional string workflowID
+  30: optional string runID
+  40: optional CrossClusterTaskType taskType
+  50: optional i16 taskState
+  60: optional i64 (js.type = "Long") taskID
+  70: optional i64 (js.type = "Long") version
+  80: optional i64 (js.type = "Long") visibilityTimestamp
+}
+
+struct CrossClusterStartChildExecutionRequestAttributes {
```

Comment:
Another option for those RequestAttributes is to reuse the Request struct we defined for history Start/Signal/Cancel workflow API. But the problem with that is 1: it assumes how tasks are processed in the target cluster (just calling the corresponding API) 2: those history request structs contain some redundant fields which are not required for processing the task.